### ISSUE TITLE
Add support for CJS imports

### ISSUE THREAD

**Issue by miketromba** [planetscale/database-js#122]
See issue: https://github.com/planetscale/database-js/issues/100
Specifically, my comment here: https://github.com/planetscale/database-js/issues/100#issuecomment-1627750733

For many typescript users, especially those using `ts-node` in their environment, a CJS export of this package would be a huge improvement to the DX.

Here is a great article from a maintainer who solved the same problem within his package (supporting both ESM and CJS since most users are still using CJS). He includes specifics of how to implement this setup: https://evertpot.com/universal-commonjs-esm-typescript-packages/

TLDR: Within `package.json`, you can specify both a CJS export and an ESM export so that users automatically get the right version of this library for their environment when they import it.

### E.g.
```json
{
  "type": "module",
  "exports": {
    "require": "./cjs/index.js",
    "import": "./esm/index.js"
  },
  "main": "cjs/index.js"
}
```

> When specifying your package.json this way, Node will automatically load the correct (cjs/esm) directory depending on what the user needs. This all happens automatically. Neat!

**Comment by miketromba** [comment]
Created PR #123

**Comment by binaryartifex** [comment]
for the love of god yes. Im categorically blocked with my app until this is sorted. I have a barely instantiated nrwl/nx monorepo with a nestjs application and a nodejs library that builds a drizzle schema and exports a planetscale connection. Everything i do to fix it leads back to this ->

```
F:\...\product-suite\node_modules\.pnpm\@nx+js@16.5.2_@swc-node+register@1.4.2_@swc+core@1.3.69_nx@16.5.2_typescript@5.1.6\node_modules\@nx\js\src\executors\node\node-with-require-overrides.js:18
        return originalLoader.apply(this, arguments);
                              ^
Error [ERR_REQUIRE_ESM]: require() of ES Module F:\...\product-suite\node_modules\.pnpm\@planetscale+database@1.8.0\node_modules\@planetscale\database\dist\index.js from F:\...\haven-world-tours\product-suite\dist\apps\web-server\main.js not supported.
Instead change the require of index.js in F:\client-projects\haven-world-tours\product-suite\dist\apps\web-server\main.js to a dynamic import() which is available in all CommonJS modules.
```

**Comment by miketromba** [comment]
@binaryartifex For now, the temporary solution I'm using is to build the package locally and store it in my codebase. You can do the same by cloning this fork (https://github.com/miketromba/database-js-cjs/tree/main) and running `npm install && npm build` Then, grab the built `dist` folder, rename it to e.g. `planetscale-database-js`, add it to your codebase, and import the library from there.

**Comment by iheanyi** [comment]
Hey y'all! Sorry for the delay in replying to this one. I'm hoping to get some time this week to review and come to a decision on CommonJS.

For my own understanding, what's preventing the use of ESM in certain projects? Looking at `ts-node`, I see they have support (although it's marked as experimental) for ESM modules, so I don't know why our driver won't work as-is today. I would love some additional context here about how it's being used to better understand, thanks!

**Comment by binaryartifex** [comment]
@miketromba is there any reason why #123 came to a grinding halt? seems like a no brainer merge?

**Comment by binaryartifex** [comment]
so are we merging #123? i think there's plenty enough argument to support it

**Comment by iheanyi** [comment]
Hey y'all, as a heads up, a new release has been cut!
